daffodil-core/src/main/scala/org/apache/daffodil/dsom/Term.scala

```diff
@@ -230,6 +230,7 @@ trait Term
         Assert.invariant(this.isInstanceOf[Root])
         None
       }
+      case gcgd: GlobalChoiceGroupDef => Some(gcgd.groupRef.asModelGroup)
```

Comment:
I imagine we also need a GlobalSequenceGroupDef for the case where there are nested sequences? Perhaps this can just be GlobalGroupDef, which is the parent class of Sequence and Choice GroupDefs? But actually, that case already exists just a few lines down. Maybe the bug is on that line, and that line should be changed to the ``Some(...)`` that you have here instead of getting the immediately enclosing model group?

Reply:
I just tested it out and this fixes the issue and doesn't seem to cause any issues in our test suite, so that works for me.

Reply:
What you fixed is that this was providing not the enclosing model group, but 2-enclosings out, in the case where the term was the model-group of a global group definition.

The same bug is here for sequences.

Change the line below for parent is a GlobalGroupDef: 

  case ggd: GlobalGroupDef => Some(ggd.groupRef.asModelGroup)

This subsumes your change line, so you can remove that.

Reply:
Ooops. Exactly same as what SL said. My job is at the redundancy dept. is where I work.